src/index/tree.rs

```diff
@@ -27,6 +28,11 @@ impl Tree {
         }
     }
 
+    fn file(&self, name: &str) -> io::Result<File> {
+        let path = self.compute_record_path(name);
+        File::create(path)
+    }
+
     pub fn match_record(&self, name: &str, req: VersionReq) -> Result<CrateVersion, Error> {
```

Comment:
This could also be implemented with `File`:
```
self.file(name)
    .records()
    .iter()
    .filter(|(vers, _)| req.matches(vers))
    .max_by(|(v1, _), (v2, _)| v1.cmp(v2))
    .map(|(_, record)| record.clone())
    .ok_or_else(|| AlexError::CrateNotFound {
        name: String::from(name),
    })
```

Reply:
i'll take a look at this. my inclination would be to push this down to the `File`. So the tree would be responsible for checking the name exists, and the file for doing the version matching.

also it seems like a mistake to clone in the method. better to return the reference, and let the caller clone it (if they need to)

Reply:
quick question- what are you using the `match record` method for?

I had considered this 'dependency resolution' task to be `Cargo`'s job